### DISCUSSION TITLE
Unable to deploy or promote to production

### DISCUSSION THREAD

**Discussion by steveharman** [planetscale/discussion#98] [Q&A] [answered]
Hi.  

Im experimenting with Planetscale and have a db with two branches, `main` and `staging`.  `main` has never been deployed to, has never had anything in it and is entirely empty.  All seems fine with my schema on staging and I'd now like to deploy its contents into production on `main`

However when I submit a deploy request I get the message "_This deploy request is not deployable_" and the long string shown at the bottom of this message.   So instead I attempted to `promote` the staging db straight to production, by visiting the staging branch and selecting "_Promote to production branch_".  Sadly this gives me a "Not found" popup.

<img width="275" alt="Screenshot_2021-09-09_at_13_12_14" src="https://user-images.githubusercontent.com/12557929/132683932-91a47dc7-97f7-475c-a2af-4703a7a2bb68.png">

I'm lost.  :-) Does anyone have any ideas please?

Thanks,

`error loading logical schema: found 8 failed statements in the schema: Statement[0]: "CREATE TABLE `wp_comments` (\n `comment_ID` bigint unsigned NOT NULL AUTO_INCREMENT,\n `comment_post_ID` bigint unsigned NOT NULL DEFAULT '0',\n `comment_author` tinytext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `comment_author_email` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `comment_author_url` varchar(200) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `comment_author_IP` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `comment_date` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `comment_date_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `comment_content` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `comment_karma` int NOT NULL DEFAULT '0',\n `comment_approved` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '1',\n `comment_agent` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `comment_type` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'comment',\n `comment_parent` bigint unsigned NOT NULL DEFAULT '0',\n `user_id` bigint unsigned NOT NULL DEFAULT '0',\n PRIMARY KEY (`comment_ID`),\n KEY `comment_post_ID` (`comment_post_ID`),\n KEY `comment_approved_date_gmt` (`comment_approved`,`comment_date_gmt`),\n KEY `comment_date_gmt` (`comment_date_gmt`),\n KEY `comment_parent` (`comment_parent`),\n KEY `comment_author_email` (`comment_author_email`(10))\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[0]: <buffer contents>:56:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'comment_date' Statement[1]: "CREATE TABLE `wp_actionscheduler_logs` (\n `log_id` bigint unsigned NOT NULL AUTO_INCREMENT,\n `action_id` bigint unsigned NOT NULL,\n `message` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `log_date_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `log_date_local` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n PRIMARY KEY (`log_id`),\n KEY `action_id` (`action_id`),\n KEY `log_date_gmt` (`log_date_gmt`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[1]: <buffer contents>:37:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'log_date_gmt' Statement[2]: "CREATE TABLE `wp_actionscheduler_actions` (\n `action_id` bigint unsigned NOT NULL AUTO_INCREMENT,\n `hook` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `status` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `scheduled_date_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `scheduled_date_local` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `args` varchar(191) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci DEFAULT NULL,\n `schedule` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci,\n `group_id` bigint unsigned NOT NULL DEFAULT '0',\n `attempts` int NOT NULL DEFAULT '0',\n `last_attempt_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `last_attempt_local` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `claim_id` bigint unsigned NOT NULL DEFAULT '0',\n `extended_args` varchar(8000) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci DEFAULT NULL,\n PRIMARY KEY (`action_id`),\n KEY `hook` (`hook`),\n KEY `status` (`status`),\n KEY `scheduled_date_gmt` (`scheduled_date_gmt`),\n KEY `args` (`args`),\n KEY `group_id` (`group_id`),\n KEY `last_attempt_gmt` (`last_attempt_gmt`),\n KEY `claim_id` (`claim_id`),\n KEY `claim_id_status_scheduled_date_gmt` (`claim_id`,`status`,`scheduled_date_gmt`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[2]: <buffer contents>:1:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'scheduled_date_gmt' Statement[3]: "CREATE TABLE `wp_posts` (\n `ID` bigint unsigned NOT NULL AUTO_INCREMENT,\n `post_author` bigint unsigned NOT NULL DEFAULT '0',\n `post_date` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `post_date_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `post_content` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `post_title` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `post_excerpt` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `post_status` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'publish',\n `comment_status` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'open',\n `ping_status` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'open',\n `post_password` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `post_name` varchar(200) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `to_ping` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `pinged` text CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `post_modified` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `post_modified_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `post_content_filtered` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `post_parent` bigint unsigned NOT NULL DEFAULT '0',\n `guid` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `menu_order` int NOT NULL DEFAULT '0',\n `post_type` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'post',\n `post_mime_type` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `comment_count` bigint NOT NULL DEFAULT '0',\n PRIMARY KEY (`ID`),\n KEY `post_name` (`post_name`(191)),\n KEY `type_status_date` (`post_type`,`post_status`,`post_date`,`ID`),\n KEY `post_parent` (`post_parent`),\n KEY `post_author` (`post_author`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[3]: <buffer contents>:1659:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'post_date' Statement[4]: "CREATE TABLE `wp_duplicator_pro_packages` (\n `id` bigint unsigned NOT NULL AUTO_INCREMENT,\n `name` varchar(250) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `hash` varchar(50) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `status` int NOT NULL,\n `created` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `owner` varchar(60) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `package` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n PRIMARY KEY (`id`),\n KEY `hash` (`hash`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[4]: <buffer contents>:86:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'created' Statement[5]: "CREATE TABLE `wp_actionscheduler_claims` (\n `claim_id` bigint unsigned NOT NULL AUTO_INCREMENT,\n `date_created_gmt` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n PRIMARY KEY (`claim_id`),\n KEY `date_created_gmt` (`date_created_gmt`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[5]: <buffer contents>:25:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'date_created_gmt' Statement[6]: "CREATE TABLE `wp_links` (\n `link_id` bigint unsigned NOT NULL AUTO_INCREMENT,\n `link_url` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_name` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_image` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_target` varchar(25) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_description` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_visible` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT 'Y',\n `link_owner` bigint unsigned NOT NULL DEFAULT '1',\n `link_rating` int NOT NULL DEFAULT '0',\n `link_updated` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `link_rel` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `link_notes` mediumtext CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL,\n `link_rss` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n PRIMARY KEY (`link_id`),\n KEY `link_visible` (`link_visible`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[6]: <buffer contents>:168:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'link_updated' Statement[7]: "CREATE TABLE `wp_users` (\n `ID` bigint unsigned NOT NULL AUTO_INCREMENT,\n `user_login` varchar(60) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_pass` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_nicename` varchar(50) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_email` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_url` varchar(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_registered` datetime NOT NULL DEFAULT '0000-00-00 00:00:00',\n `user_activation_key` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n `user_status` int NOT NULL DEFAULT '0',\n `display_name` varchar(250) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL DEFAULT '',\n PRIMARY KEY (`ID`),\n KEY `user_login_key` (`user_login`),\n KEY `user_nicename` (`user_nicename`),\n KEY `user_email` (`user_email`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;\n" Error[7]: <buffer contents>:1734:1: Error executing DDL in workspace: Error 1067: Invalid default value for 'user_registered'`

**Comment by jgreet** [comment, marked as answer]
Hi @steveharman, I think two things are going wrong here.

First, branch promotion shouldn't have been available yet. Looks like we leaked an upcoming feature there 😉 

Second, this type of error appears several times in the deploy request message:

`Error executing DDL in workspace: Error 1067: Invalid default value for 'user_registered'`

`user_registereddatetime NOT NULL DEFAULT '0000-00-00 00:00:00',`

This is essentially the same issue as https://github.com/planetscale/beta/discussions/62. It's likely `NO_ZERO_DATE` or strict mode was unset in the session where those tables were created. The solution is to use null or another dummy default date. 

Sorry about the confusing error message, we can do a better job of catching this earlier and explaining the issue in the app.

**Reply by steveharman** [reply]
That makes sense, although there isn't much I can do about the schema from my side.  It's really frustrating - everything is working perfectly in the staging branch I created, I just can't move it into production.   

I played with dumping & restoring the db between staging > main but that throws an EOF when I try to restore - even dumping staging & restoring back to staging agains does the same.

So close, and yet so far.   :-)

Thanks for your reply though.
